scripts/deactivate_iac.py

```diff
@@ -0,0 +1,101 @@
+from structlog import configure, wrap_logger
+from structlog.processors import TimeStamper, JSONRenderer
+from structlog.stdlib import add_log_level, filter_by_level, LoggerFactory
+
+try:
+    config_info = getattr(config, os.environ['APP_SETTINGS'])
+except (AttributeError, KeyError) as e:
+    config_info = config.DevelopmentConfig
+
+# Put config into a dict
+config = dict((name, getattr(config_info, name)) for name in dir(config_info) if not name.startswith('__'))
+0
+# Configure logger
+logger = wrap_logger(logging.getLogger(__name__))
+
+# Configure argument parser
+parser = argparse.ArgumentParser(description='Deactivate IACs for a Collection Exercise.')
+parser.add_argument('-c', dest='collectionExercise', type=str, help='Collection Exercise ID.')
+
+# Setup URLs
+case_url = f"{config['CASE_URL']}/cases/"
+iac_url = f"{config['IAC_URL']}/iacs/"
+ci_url = f"{config['COLLECTION_INSTRUMENT_URL']}/collection-instrument-api/1.0.2/collectioninstrument/id/"
+collex_url = f"{config['COLLECTION_EXERCISE_URL']}/collectionexercises/"
+sample_url = f"{config['SAMPLE_URL']}/samples/"
+eq_url = f"{config['EQ_URL']}"
+
+
+def main(collection_ex_id):
+
+    collection_ex_info = requests.get(collex_url + "link/" + collection_ex_id,
+                                      auth=config["COLLECTION_EXERCISE_AUTH"])
+    collection_ex_info.raise_for_status()
+    sample_summary_id = collection_ex_info.json()
+    sample_units = requests.get(sample_url + sample_summary_id[0] + "/sampleunits",
+                                auth=config["SAMPLE_AUTH"])
+    sample_units = sample_units.json()
+    samples = [sample['id'] for sample in sample_units]
+    logger.info(sample_size=len(samples))
+
+    sample_chunks = [samples[i:i + 10] for i in range(0, len(samples), 10)]
+    sample_return = []
+    for sample_chunk in sample_chunks:
+        sample_chunk_return = requests.get(case_url + "sampleunitids?sampleUnitId=" + ','.join(sample_chunk),
+                                           auth=config["CASE_AUTH"])
+        sample_chunk_return.raise_for_status()
+        for sample in sample_chunk_return.json():
+            sample_return.append(sample)
+
+    for sample in sample_return:
+        if sample["sampleUnitType"] == "H" and sample["state"] == "INACTIONABLE":
+            case_id = sample["id"]
+            iac_return = requests.get(case_url + case_id + "/iac", auth=config["CASE_AUTH"])
+            iac_return.raise_for_status()
+            iacs = iac_return.json()
+            for iac in iacs:
+                iac = requests.get(iac_url + iac['iac'], auth=config["IAC_AUTH"])
+                iac.raise_for_status()
+                iac_data = iac.json()
+                logger.info('Found IAC', iac=iac_data["iac"],  active=iac_data["active"])
+                if iac_data.get("active", False):
+                    deactivate_iac(iac.json())
+
+
+def deactivate_iac(iac):
+    deactivate_data = {"active": "false", "updatedBy": "Tricky"}
```

Comment:
What does updated by Tricky mean?

Reply:
Changed to SYSTEM.

Reply:
Could we not have something descriptive like "RM_DEACTIVE_IAC_SCRIPT" to make the audit trail a bit clearer?